### ISSUE TITLE
Compiling error : HWLOC_OBJ_NODE

### ISSUE THREAD

**Issue by teicors** [snort3/snort3#350]
Hi all,
during the compiling' the git version "master", I achieve this error on Ubuntu 22.04 LTS:

```
Consolidate compiler generated dependencies of target main
[ 13%] Building CXX object src/main/CMakeFiles/main.dir/thread_config.cc.o
/home/angelo/Software/snort3/src/main/thread_config.cc: In member function ‘int snort::ThreadConfig::get_numa_node(hwloc_topology_t, hwloc_cpuset_t)’:
/home/angelo/Software/snort3/src/main/thread_config.cc:238:49: error: ‘HWLOC_OBJ_NODE’ was not declared in this scope; did you mean ‘HWLOC_OBJ_CORE’?
  238 |     int depth = hwloc->get_type_depth(topology, HWLOC_OBJ_NODE);
      |                                                 ^~~~~~~~~~~~~~
      |                                                 HWLOC_OBJ_CORE
make[2]: *** [src/main/CMakeFiles/main.dir/build.make:280: src/main/CMakeFiles/main.dir/thread_config.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3334: src/main/CMakeFiles/main.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

Using the suggestion, I catch the 100%m, but I'm not really confident: any suggestion ?

Thansk,
Angelo

PS I can compile flawless the 3.1.28 and 3.1.48

**Comment by ansrb** [comment]
Hey. Thanks for reporting this, it looks like `HWLOC_OBJ_NODE` has been deprecated for a long time and has been recently removed altogether from hwloc: https://github.com/open-mpi/hwloc/commit/058f1114f441078696058e835ebd11930e43c804
We will update this.

I suppose you are using hwloc library from master branch. As a temporary workaround for this build issue, you could try switching to their latest tagged release, I see that 2.10 still supports this deprecated macro.

**Comment by ansrb** [comment]
This update will be available in the next open source release.

**Comment by InverseRE** [comment]
Fixed in 8f8e9cf28856359a1ef3081baa9240f90276f8d4
